grr/server/grr_response_server/gui/api_plugins/metadata.py

```diff
@@ -804,6 +804,10 @@ def _GetPaths(self) -> Dict[str, Dict]:
       ungrouped_routes = []
       for http_method, path, strip_root_types in router_method.http_methods:
         path_components = path.split("/")
+        # Remove trailing empty strings from the list of path components.
+        while path_components and len(path_components[-1]) == 0:
+          path_components.pop()
```

Comment:
Do we need to only remove trailing slashes? If we would filter all non-empty strings, wouldn't it be better? Empty path components do not have any meaning anyway, do they?

Reply:
You're right! Currently there is one empty string in all lists caused by the root "/" when calling `str.split`, sort of representing the root component, but it doesn't help us in any way and is present for all routes so it shouldn't affect the path argument extraction. I'll filter out all the empty strings. Thank you!

Reply:
Done. I needed to add the root "/" in `_NormalizePath` because previously it was added using `"/".join(...)` relying on that empty string caused by the `str.split("/")`.